Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1

```diff
@@ -1,296 +1,122 @@
-function Get-RoleName($extPath)
-{
-    $roleName = ""
+    $diagnosticExtensions = Get-DiagnosticsExtensions $StorageAccount $serviceConfigFile
 
-    #The following statement uses the SimpleMatch option to direct the -split operator to interpret the dot (.) delimiter literally.
-    #With the default, RegexMatch, the dot enclosed in quotation marks (".") is interpreted to match any character except for a newline
-    #character. As a result, the Split statement returns a blank line for every character except newline.  The 0 represents the "return
-    #all" value of the Max-substrings parameter. You can use options, such as SimpleMatch, only when the Max-substrings value is specified.
-    $roles = $extPath -split ".",0,"simplematch"
-
-    if ($roles -is [system.array] -and $roles.Length -gt 1)
-    {
-        $roleName = $roles[1] #base role name
+    $label = $DeploymentLabel
 
-        $x = 2
-        while ($x -le $roles.Length)
-        {
-            if ($roles[$x] -ne "PubConfig")
-            {
-                $roleName = $roleName + "." + $roles[$x]
-            }
-            else
-            {
-                break
-            }
-            $x++
-        }
-    }
-    else
+    if ($label -and $appendDateTime)
     {
-        Write-Warning (Get-LocalizedString -Key "'{0}' could not be parsed into parts for registering diagnostics extensions." -ArgumentList $extPath)
+	    $label += " "
+	    $label += Get-Date
     }
 
-    return $roleName
-}
+    Write-Host "Get-AzureDeployment -ServiceName $ServiceName -Slot $Slot -ErrorAction SilentlyContinue -ErrorVariable azureDeploymentError"
+    $azureDeployment = Get-AzureDeployment -ServiceName $ServiceName -Slot $Slot -ErrorAction SilentlyContinue -ErrorVariable azureDeploymentError
 
-function Get-DiagnosticsExtensions($storageAccount, $extensionsPath)
-{
-    $diagnosticsConfigurations = @()
-    
-    $extensionsSearchPath = Split-Path -Parent $extensionsPath
-    Write-Verbose "extensionsSearchPath= $extensionsSearchPath"
-    $extensionsSearchPath = Join-Path -Path $extensionsSearchPath -ChildPath "Extensions"
-    Write-Verbose "extensionsSearchPath= $extensionsSearchPath"
-    #$extensionsSearchPath like C:\Agent\_work\bd5f89a2\staging\Extensions
-    if (!(Test-Path $extensionsSearchPath))
+    if($azureDeploymentError) {
+       $azureDeploymentError | ForEach-Object { Write-Warning $_.Exception.ToString() }
+    }
+
+    if (!$azureDeployment)
     {
-        Write-Verbose "No Azure Cloud Extensions found at '$extensionsSearchPath'"
+	    if ($label)
+	    {
+		    Write-Host "New-AzureDeployment -ServiceName $ServiceName -Package $servicePackageFile -Configuration $serviceConfigFile -Slot $Slot -Label $label -ExtensionConfiguration <extensions>"
+		    $azureDeployment = New-AzureDeployment -ServiceName $ServiceName -Package $servicePackageFile -Configuration $serviceConfigFile -Slot $Slot -Label $label -ExtensionConfiguration $diagnosticExtensions
```

Comment:
It looks like you could combine the four separate calls to New-AzureDeployment into one by leverage the splat feature of PowerShell. To format the "##command" line, look at the Format-Splat function in Common/VstsAzureHelpers_